### PR TITLE
test: testground sanity check with knuu

### LINKED ISSUE #3189: Running txsim as a knuu instance inside e2e tests
This issue is to add the capability to execute txsim as a knuu instance within e2e tests, enhancing efficiency and reliability when loading the network with a high volume of transactions.

Part of #3147

### PR DESCRIPTION
**Pull request by staheri14**
In replace of https://github.com/celestiaorg/celestia-app/pull/3184 
Closes #3189
- Introduces the capability to customize the resource allocation for each Node and TxSimNode individually, allowing for a diverse range of resource specifications rather than applying a uniform resource allocation across all instances.
- Enables running txsim as a knuu instance.

### CODE REVIEW

File: test/e2e/testnet.go [hunk 1]
```diff
@@ -18,6 +18,7 @@ type Testnet struct {
 	nodes           []*Node
 	genesisAccounts []*Account
 	keygen          *keyGenerator
+	txSimNode       *TxSim
```
Comment (cmwaters) [inline test/e2e/testnet.go:21]: Do we want just a single tx sim node? Would that be sufficient to load the testnet?

Maybe it's fine to start with one and not prematurely over-engineer it

File: test/e2e/testnet.go [hunk 2]
```diff
@@ -178,6 +240,12 @@ func (t *Testnet) Cleanup() {
 			log.Err(err).Msg(fmt.Sprintf("node %s failed to cleanup", node.Name))
 		}
 	}
+	if t.txSimNode.Instance != nil {
+		err := t.txSimNode.Instance.Destroy()
+		if err != nil {
+			log.Err(err).Msg(fmt.Sprintf("txsim %s failed to cleanup", t.txSimNode.Name))
+		}
+	}
```
Comment (cmwaters) [inline test/e2e/testnet.go:248]: I think we need to stop the instance first and then destroy it

File: test/e2e/testnet.go [hunk 3]
```diff
@@ -34,32 +35,92 @@ func New(name string, seed int64) (*Testnet, error) {
...
+	resources Resources,
+	volumePath string,
+	keyringPath string,
+) error {
+	txsim, err := CreateTxSimNode(name, version, endpoint, seed, sequences,
+		blobRange, pollTime, resources, volumePath)
+	if err != nil {
+		return err
+	}
+	err = txsim.Instance.AddFolder(keyringPath, volumePath, "10001:10001")
```
Comment (cmwaters) [inline test/e2e/testnet.go:97]: I didn't know this was possible. This is useful also for regular consensus nodes

**Review by cmwaters — COMMENTED** [review]
Have you had any preliminary results yet as to whether having it as a separate instance enables greater load on the network?

File: test/e2e/testnet.go (hunk 1 above)
Comment (staheri14) [reply test/e2e/testnet.go:21]: >Maybe it's fine to start with one and not prematurely over-engineer it

Indeed, I intended to do it incrementally, first by testing out a single instance of txsim and debugging it, then scaling up if the desired load is not met.

File: test/e2e/testnet.go (hunk 2 above)
Comment (staheri14) [reply test/e2e/testnet.go]: Good point, added this https://github.com/celestiaorg/celestia-app/pull/3194/commits/131e021dd7840190bce67a3743f562812d7bb905

**Comment by staheri14** [comment]
> Have you had any preliminary results yet as to whether having it as a separate instance enables greater load on the network?

Not yet, was awaiting reviews on the PR before moving forward with the test. I will conduct it and share the results soon.

File: test/e2e/txsimNode.go [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+	}
+	err = instance.SetMemory(resources.memoryRequest, resources.memoryLimit)
+	if err != nil {
+		return nil, err
+	}
+	err = instance.SetCPU(resources.cpu)
+	if err != nil {
+		return nil, err
+	}
+	err = instance.AddVolumeWithOwner(volumePath, resources.volume, 10001)
```
Comment (smuu) [inline test/e2e/txsimNode.go:48]: I don't know if this is needed for txsim. `AddVolume()` is required if you want data persistence after a restart.
As txsim produces no local data and you do not intend to restart it, I think you can safely remove this.

**Review by smuu — COMMENTED** [review]
This looks great :rocket: 

I only have two small things

File: test/e2e/txsimNode.go (hunk 4 above)
Comment (cmwaters) [reply test/e2e/txsimNode.go:48]: What about for upgrades. Doesn't that involve restarting the instance?

File: test/e2e/testnet.go [hunk 5]
```diff
@@ -34,32 +37,135 @@ func New(name string, seed int64) (*Testnet, error) {
...
+		return err
+	}
+
+	t.txSimNodes = append(t.txSimNodes, txsim)
+	return nil
+}
+
+func (t *Testnet) StartTxSimNodes() error {
+	for _, txsim := range t.txSimNodes {
+		err := txsim.Instance.Start()
```
[the remark this answers is not in the document]
Comment (staheri14) [reply test/e2e/testnet.go:128]: Thanks! I have integrated your comment in my other [PR ](https://github.com/celestiaorg/celestia-app/pull/3244)

File: test/e2e/txsimNode.go (hunk 4 above)
Comment (staheri14) [reply test/e2e/txsimNode.go:48]: Thank you! Following your suggestion, I removed it, and everything was fine. I've also integrated your suggestion into my other PR [here](https://github.com/celestiaorg/celestia-app/pull/3244).

**Comment by staheri14** [comment]
Migrating this PR to this [one](https://github.com/celestiaorg/celestia-app/pull/3244)

**Pull request closed without merging**